devU-api/src/containerAutoGrader/containerAutoGrader.service.ts

```diff
@@ -0,0 +1,74 @@
+import { getRepository, IsNull } from 'typeorm'
+
+import { ContainerAutoGrader } from '../../../devu-shared/src'
+
+import ContainerAutoGraderModel from './containerAutoGrader.model'
+import { minioClient, BucketNames } from '../fileStorage'
+
+const connect = () => getRepository(ContainerAutoGraderModel)
+
+function assignmentGraderFileRecordName(containerAutoGrader: ContainerAutoGrader) {
+  if (!containerAutoGrader.id) throw new Error('Missing Id')
+  return containerAutoGrader.id.toString()
+}
+
+export async function create(containerAutoGrader: ContainerAutoGrader, graderInputFile: Buffer, makefileInputFile: Buffer | null = null) {
+    containerAutoGrader.graderFilename = "Temp value. You'll see this if the file upload to MinIO fails"
+    const newContainerAutoGrader = await connect().save(containerAutoGrader)
+    const graderFileRecordName: string = assignmentGraderFileRecordName(newContainerAutoGrader)
+
+    /*
+      * For the minioClient.putObject method, I am not sure how the objectName is, but it seems like
+      * the objectName is the id of the containerAutoGrader.(inherits from the codeAssignment.service.ts file),
+      * but I am not sure if the makefile should also use the same objectName as the grader. Marking this 
+      * for review. Same with the update method.
+    */
```

Comment:
This does need to be rethought since we want to support multiple graders per assignment and this would not allow that. Kicking the can down the roan since we have to rethink file uploads completely

Reply:
To more directly answer the question, since I see you just named the Makefile "Makefile" which will cause a name conflict when a second grader is uploaded, yes the makefile should follow whatever naming convention we end up with for autograding files